src/bio/bio_buffer.c

```diff
@@ -1082,15 +1199,27 @@ bio_rw(struct bio_io_context *ioctxt, bio_addr_t addr, d_iov_t *iov,
 	sgl.sg_nr_out = 0;
 
 	rc = bio_rwv(ioctxt, &bsgl, &sgl, update);
-	if (rc)
+	mea->mea_inflights = 1;
+	mem.mem_update = update;
+	mem.mem_bs = ioctxt->bic_xs_ctxt->bxc_blobstore;
+	mem.mem_tgt_id = ioctxt->bic_xs_ctxt->bxc_xs_id;
+
+	if (rc) {
+		spdk_thread_send_msg(owner_thread(mem.mem_bs), bio_media_error,
+				     &mem);
+		/* Wait for media error event to complete */
+		media_error_completion(ioctxt->bic_xs_ctxt, mea);
+
 		D_ERROR("%s to blob:%p failed for xs:%p, rc:%d\n",
 			update ? "Write" : "Read", ioctxt->bic_blob,
 			ioctxt->bic_xs_ctxt, rc);
-	else
+	} else
 		D_DEBUG(DB_IO, "%s to blob %p for xs:%p successfully\n",
 			update ? "Write" : "Read", ioctxt->bic_blob,
 			ioctxt->bic_xs_ctxt);
 
+	media_error_cp_arg_fini(mea);
+
```

Comment:
Instead of changing all these bio_rw/readv/writev functions, we'd just modify rw_completion() to send media error on io error, right?

Reply:
makes sense, simplifies code